### PR TITLE
feature(nyz): add policy gradient algo implementation

### PR DESCRIPTION
**Pull request by PaParaZz1**
## Description
add policy gradient (REINFORCE) algo

LunarLander result:
Screen Shot 2022-11-22 at 10 19 06 PM

## Related Issue
#517 

## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: ding/policy/pg.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        self._model.eval()
+        with torch.no_grad():
+            output = self._model.forward(data)
+            if self._cfg.deterministic_eval:
+                if self._cfg.action_space == 'discrete':
+                    output['action'] = output['logit'].argmax(dim=-1)
+                elif self._cfg.action_space == 'continuous':
+                    output['action'] = output['logit']['mu']
+                else:
+                    raise KeyError("invalid action_space: {}".format(self._cfg.action_space))
```
Comment (karroyan) [inline ding/policy/pg.py:201]: Why key error here, but runtime error in ding/model/template/pg.py?

File: README.md [hunk 2]
```diff
@@ -193,46 +193,47 @@ P.S: The `.py` file in `Runnable Demo` can be found in `dizoo`
...
-|  47  |         [PLR](https://arxiv.org/pdf/2010.03934.pdf)          |   ![other](https://img.shields.io/badge/-other-lightgrey)    | [PLR doc](https://di-engine-docs.readthedocs.io/en/latest/12_policies/plr.html)<br>[data/level_replay/level_sampler](https://github.com/opendilab/DI-engine/blob/main/ding/data/level_replay/level_sampler.py) |        python3 -u bigfish_plr_config.py -s 0       |
-|  48  |         [PCGrad](https://arxiv.org/pdf/2001.06782.pdf)          |   ![other](https://img.shields.io/badge/-other-lightgrey)    | [torch_utils/optimizer_helper/PCGrad](https://github.com/opendilab/DI-engine/blob/main/ding/data/torch_utils/optimizer_helper.py) |        python3 -u multi_mnist_pcgrad_main.py -s 0       |
+|  9   |         [PG](https://proceedings.neurips.cc/paper/1999/file/464d828b85b0bed98e80ade0a5c43b0f-Paper.pdf)            | ![discrete](https://img.shields.io/badge/-discrete-brightgreen) | [PG doc](https://di-engine-docs.readthedocs.io/en/latest/12_policies/a2c.html)<br>[policy/pg](https://github.com/opendilab/DI-engine/blob/main/ding/policy/pg.py) |        ding -m serial -c cartpole_pg_config.py -s 0         |
```
Comment (karroyan) [inline README.md:196]: The PG doc link is wrong

Comment (PaParaZz1) [reply README.md:196]: We don't add doc about PG, I use this link as a placeholder.

File: ding/policy/pg.py (hunk 1 above)
Comment (PaParaZz1) [reply ding/policy/pg.py:201]: I modified runtime error in model to key error. Generally speaking, we should annotate error type in as much details as possible. runtime error is just a default case.
